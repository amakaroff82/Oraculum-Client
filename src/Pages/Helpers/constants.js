import { MC_BASE_URL } from '../../Shared/constants';

