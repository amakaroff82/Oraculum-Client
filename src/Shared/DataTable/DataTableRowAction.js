import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import IconButton from 'material-ui/IconButton';

const styles = theme => ({
  root: {
    background: 'transparent',
    display: 'inline-block',
  },
  icon: {
    cursor: 'pointer',
    '&, & a': {
      color: theme.palette.text.disabled,
    },
    '.rt-tr:hover &, .rt-tr:hover & a': {
      color: theme.palette.primary[500],
    },
    '.rt-tr &:hover, .rt-tr &:hover a': {
      color: theme.palette.primary[900],
    },
  },
  iconDisabled: {
    '.rt-tr:hover &, .rt-tr:hover & a': {
      color: `${theme.palette.text.disabled}`,
    },
  },
});

export const DataTableRowAction = ({
  childComponent,
  text,
  onClick,
  classes,
  disabled,
}) => {
  if (onClick) {
    return (
      <span className={classes.root}>
        <IconButton
          onClick={onClick}
          classes={{ root: disabled ? classes.iconDisabled : classes.icon }}
          title={text}
          disabled={disabled}
        >
          {childComponent}
        </IconButton>
      </span>
    );
  } else {
    return (
      <span className={disabled ? classes.iconDisabled : classes.icon}>
        {childComponent}
      </span>
    );
  }
};

DataTableRowAction.propTypes = {
  childComponent: PropTypes.object,
  text: PropTypes.string,
  onClick: PropTypes.any,
  disabled: PropTypes.bool,
};

export default withStyles(styles, { name: 'DataTableRowAction' })(
  DataTableRowAction
);
