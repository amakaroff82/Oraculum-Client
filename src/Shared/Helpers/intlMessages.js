export const intlMessages = {
  orchestration: { id: 'Orchestration', defaultMessage: 'Orchestration' },
  products: { id: 'Products', defaultMessage: 'Products' },
  status: { id: 'Status', defaultMessage: 'Status' },
  dateLastModified: {
    id: 'DateLastModified',
    defaultMessage: 'Date last modified',
  },
  title: { id: 'Title', defaultMessage: 'Titre' },
  topic: { id: 'Topic', defaultMessage: 'Topic' },
  createdBy: { id: 'CreatedBy', defaultMessage: 'Created by' },
  version: { id: 'Version', defaultMessage: 'Version' },
};

export default intlMessages;
