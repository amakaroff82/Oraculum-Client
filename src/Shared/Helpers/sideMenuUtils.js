import React from 'react';
import { ListItem, ListItemText } from 'material-ui/List';

export const renderListItems = (listItemData, listAction, identifierKey) => {
  if (listItemData && listAction && identifierKey) {
    return listItemData.map((element, index) => {
      return (
        <ListItem
          key={element[identifierKey]}
          button
          onClick={() => listAction(element)}
        >
          <ListItemText color="primary" primary={element[identifierKey]} />
        </ListItem>
      );
    });
  }
};
