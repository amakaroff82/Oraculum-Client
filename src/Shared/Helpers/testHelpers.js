// Create mock response for testing
export const makeMockResponse = (
  response,
  status = 200,
  statusText = 'OK',
  headers = { 'Content-Type': 'application/json' }
) => {
  const init = {
    status,
    statusText,
    headers,
  };
  return new window.Response(JSON.stringify(response), init);
};

export const jsonParseError = () => {
  try {
    JSON.parse('+data+');
  } catch (err) {
    return err;
  }
};

export const makeFakeError = (
  response = null,
  status = 400,
  statusText = 'Bad Request'
) => {
  const error = new Error('Error');
  error.response = makeMockResponse(response, status, statusText);
  return error;
};
