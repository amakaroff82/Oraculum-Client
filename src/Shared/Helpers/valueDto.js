/// Create items for binding Std ValueDTOs from the web service
export const getDropDownValues = dtoArray => {
  if (!dtoArray) {
    return [];
  }
  return dtoArray.map(element => ({
    value: element.code,
    label: element.description || element.code,
  }));
};

export const getAutocompleteValues = dtoArray => {
  if (!dtoArray) {
    return [];
  }
  var result = {};
  for (var value of dtoArray) {
    result[value['code']] = value['description'] || value['code'];
  }
  return result;
};
