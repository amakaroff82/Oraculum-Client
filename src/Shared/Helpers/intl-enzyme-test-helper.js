/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { intlMessages } from './intlMessages';

