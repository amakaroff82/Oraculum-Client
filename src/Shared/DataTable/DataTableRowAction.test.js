import * as React from 'react';
import { shallowWithIntl } from '../../Shared/Helpers/intl-enzyme-test-helper';
import toJson from 'enzyme-to-json';
import { DataTableRowAction } from './DataTableRowAction';

const defaultProps = {
  classes: {},
  childComponent: <div />,
  text: '',
  onClick: () => {},
};

const setup = setupProps => {
  const props = { ...defaultProps, ...setupProps };
  return shallowWithIntl(<DataTableRowAction {...props} />);
};

describe('<DataTableRowAction />', () => {
  it('renders snapshot correctly when onClick prop is present', () => {
    const tree = setup();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders snapshot correctly when onClick prop isnt present', () => {
    const altProps = {
      onClick: null,
    };
    const tree = setup(altProps);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
