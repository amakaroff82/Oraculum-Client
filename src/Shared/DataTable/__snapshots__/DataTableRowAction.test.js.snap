// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTableRowAction /> renders snapshot correctly when onClick prop is present 1`] = `
<span>
  <withStyles(IconButton)
    classes={
      Object {
        "root": undefined,
      }
    }
    onClick={[Function]}
    title=""
  >
    <div />
  </withStyles(IconButton)>
</span>
`;

exports[`<DataTableRowAction /> renders snapshot correctly when onClick prop isnt present 1`] = `
<span>
  <div />
</span>
`;
